File: src/backend/MyNovelBuilder/MyNovelBuilder.WebApi/Dtos/Compendium/CompendiumDto.cs
namespace MyNovelBuilder.WebApi.Dtos.Compendium;

/// <summary>
/// Data transfer object for a compendium.
/// </summary>
public class CompendiumDto
{
    /// <summary>
    /// The date and time the compendium was created.
    /// </summary>
    public required DateTime CreatedAt { get; set; }
    
    /// <summary>
    /// The date and time the compendium was last updated.
    /// </summary>
    public required DateTime UpdatedAt { get; set; }
    
    /// <summary>
    /// The compendium's name.
    /// </summary>
    public required string Name { get; set; }

    /// <summary>
    /// The compendium's description.
    /// </summary>
    public required string Description { get; set; }
}

File: src/frontend/my-novel-builder/src/app/types/dtos/compendium/compendium.dto.ts
import { CompendiumRecordOverviewDto } from '../compendium-record/compendium-record-overview.dto';

export interface CompendiumDto {
  id: string;
  createdAt: string;
  updatedAt: string;
  name: string;
  description: string;
  records: CompendiumRecordOverviewDto[];
}
